### PR TITLE
Add link button to share data space

### PR DESCRIPTION
**Pull request by travisstebbins**
## Summary

Add a button to share the current data space setup (data space, execution context, runtime, class).

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

File: packages/legend-extension-dsl-data-space/src/components/query-builder/DataSpaceQueryBuilder.tsx [hunk 1]
```diff
@@ -225,10 +228,55 @@ const DataSpaceQueryBuilderSetupPanelContent = observer(
...
+            runtimePath,
+            queryBuilderState.class?.path,
+          ),
+        );
+
+      navigator.clipboard
+        .writeText(route)
+        .catch(() =>
+          applicationStore.notificationService.notifyError(
+            'Error copying data space link to clipboard',
```
Comment (YannanGao-gs) [inline packages/legend-extension-dsl-data-space/src/components/query-builder/DataSpaceQueryBuilder.tsx:261]: Do you think if we could change `data space link` to something like `data space query set up link`

File: packages/legend-extension-dsl-data-space/src/components/query-builder/DataSpaceQueryBuilder.tsx [hunk 2]
```diff
@@ -225,10 +228,55 @@ const DataSpaceQueryBuilderSetupPanelContent = observer(
...
+        'Copied data space link to clipboard',
+      );
+    };
+
     return (
       <div className="query-builder__setup__config-group">
         <PanelHeader title="properties">
           <PanelHeaderActions>
+            <PanelHeaderActionItem
+              title="copy data space link to clipboard"
```
Comment (YannanGao-gs) [inline packages/legend-extension-dsl-data-space/src/components/query-builder/DataSpaceQueryBuilder.tsx:275]: do you think if we should disable when there is no dataspace selected in the dataspace dropdown?

File: packages/legend-extension-dsl-data-space/src/components/query-builder/DataSpaceQueryBuilder.tsx (hunk 1 above)
Comment (travisstebbins) [reply packages/legend-extension-dsl-data-space/src/components/query-builder/DataSpaceQueryBuilder.tsx]: Done!

File: packages/legend-extension-dsl-data-space/src/components/query-builder/DataSpaceQueryBuilder.tsx (hunk 2 above)
Comment (travisstebbins) [reply packages/legend-extension-dsl-data-space/src/components/query-builder/DataSpaceQueryBuilder.tsx]: Done!

**Review by YannanGao-gs — APPROVED** [review]

**Pull request merged**